src/pages/index.js

```diff
@@ -6,121 +6,415 @@ import Section from "../components/Section.js";
 
 const editProfilePopup = new PopupWithForm(
   selectors.editProfileModal,
   handleEditProfileFormSubmit
 );
-//editProfilePopup.setEventListeners();
-
-console.log(editProfileModal);
 
 const editUserInfo = new UserInfo({
   titleSelector: ".profile__title",
   occupationSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
 const addNewCardPopup = new PopupWithForm(
   selectors.addNewCardModal,
   handleAddCardFormSubmit
 );
-console.log(addNewCardModal);
+
+const updateAvatarPopup = new PopupWithForm(
+  selectors.updateAvatarModal, 
+  handleProfileUpdateAvatarFormSubmit
+);
 
 const addNewCardFormValidator = new FormValidator(config, addCardForm);
 addNewCardFormValidator.enableValidation();
 
+const updateAvatarFormValidator = new FormValidator(config, updateAvatarForm);
+updateAvatarFormValidator.enableValidation();
+
+
+//USERS
+
+//API instances
+
+//User routes
+
+// GET /users/me – Get the current user’s info
+// PATCH /users/me – Update your profile information
+// PATCH /users/me/avatar – Update avatar
+// Card routes
+
+// GET /cards – Get all cards
+// POST /cards – Create a card
+// DELETE /cards/:cardId – Delete a card
+// PUT /cards/:cardId/likes – Like a card
+// DELETE /cards/:cardId/likes – Dislike a card
+
+
+//User Profile 
+
+let userInfo;
+
+
+// TODO delete
+
+api
+  .getUserInfo() //fetch info using api.getUserInfo()
+  .then((result) => {
+    userInfo = result; //store info in const
+    console.log(userInfo);
+    return api.getInitialCards();
+  })
+  .then((initialCards) => {
+    console.log(initialCards);
+    // //console.log(cardData);
+    // return userInfo._id;
+  })
+  // .then((card) => {
+  //   console.log(card);
+  //   console.log()
+  // })
+  .catch((err) => {
+    console.error(err); 
+    console.error("")
+  });
+
+  
+  api.getUserInfo()
+    .then((cardData) => {
+    editUserInfo.setUserAvatarInfo({ avatar: cardData.avatar });
+  });
+
+  
+
+api.getUserInfo()
+  .then((cardData) => {
+    editUserInfo.setUserInfo({name: cardData.name, description: cardData.about, avatar: cardData.avatar});
+  });
+
+
+
+
+  function handleEditProfileFormSubmit(cardData) {
+    editProfilePopup.submitButtonLoadingState(false); 
+  //   profileTitleInput.value = userInfo.name;   // Pre-fill with current name
+  // profileDescriptionInput.value = userInfo.about; 
+    
+    const name = profileTitleInput.value;
+    const about = profileDescriptionInput.value;
+
+    api
+    .editProfile({name, about})
+    .then((result) => {
+      console.log(result);
+      editUserInfo.setUserInfo({name: result.name, about: result.about});
+      editProfileSubmitButton.textContent = "Saving..."
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+      console.error("Changes to profile submission unsuccessful. Error."); 
+    })
+    .finally(() => {
+      editProfilePopup.submitButtonLoadingState(true); 
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```